tremor-value/src/value.rs

```diff
@@ -82,6 +85,170 @@ pub enum Value<'value> {
     Bytes(Bytes<'value>),
 }
 
+impl<'value> Eq for Value<'value> {}
+
+#[derive(PartialEq)]
+struct Static(StaticNode);
+
+impl Eq for Static {}
+
+impl PartialOrd for Static {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+impl Ord for Static {
+    // We allow this since we check bounds before we cast
+    #[allow(
+        clippy::cast_possible_truncation,
+        clippy::cast_precision_loss,
+        clippy::clippy::cast_sign_loss
+    )]
+    fn cmp(&self, other: &Self) -> Ordering {
+        match (self.0, other.0) {
+            (StaticNode::Null, StaticNode::Null) => Ordering::Equal,
+            (StaticNode::Null, _) => Ordering::Greater,
+            (_, StaticNode::Null) => Ordering::Less,
+            (StaticNode::Bool(v1), StaticNode::Bool(v2)) => v1.cmp(&v2),
+            (StaticNode::Bool(_b), _) => Ordering::Greater,
+            (_, StaticNode::Bool(_b)) => Ordering::Less,
+            (StaticNode::U64(v1), StaticNode::U64(v2)) => v1.cmp(&v2),
+            (StaticNode::U64(v1), StaticNode::I64(v2)) => {
+                if let Ok(v2) = v2.try_into() {
+                    v1.cmp(&v2)
+                } else {
+                    Ordering::Greater
+                }
+            }
+            (StaticNode::U64(v1), StaticNode::F64(v2)) => {
+                if v2 < u64::min_value() as f64 {
+                    Ordering::Greater
+                } else if v2 > u64::max_value() as f64 {
+                    Ordering::Less
+                } else {
+                    v1.cmp(&(v2 as u64))
+                }
+            }
+            (StaticNode::I64(v1), StaticNode::I64(v2)) => v1.cmp(&v2),
+            (StaticNode::I64(v1), StaticNode::U64(v2)) => {
+                if let Ok(v1) = v1.try_into() {
+                    let v1: u64 = v1;
+                    v1.cmp(&v2)
+                } else {
+                    Ordering::Less
+                }
+            }
+            (StaticNode::I64(v1), StaticNode::F64(v2)) => {
+                if v2 < i64::min_value() as f64 {
+                    Ordering::Greater
+                } else if v2 > i64::max_value() as f64 {
+                    Ordering::Less
+                } else {
+                    v1.cmp(&(v2 as i64))
+                }
+            }
+            // This is not great!
```

Comment:
why though?

Reply:
while we don't expose it but float doesn't implement Ord since it refuses to compare `Nan==Nan` which makes sense so we kind of cheat around it

Reply:
got it.

Reply:
added this as a comment